docs/concepts/storage.md

```diff
@@ -3,3 +3,157 @@ sidebar_position: 3
 ---
 
 # Storage
+
+:::tip TLDR
+
+The punchline from this concepts doc is that Responsive Platform separates
+storage from compute as a means of improving operations without sacraficing
+transactional semantics at a neglible performance penalty.
+
+:::
+
+## History & Background
```

Comment:
General comment: I feel we are spending relatively more time on the old architecture than _how_ our solution works. Maybe there isn't much to say on the latter, but this is the main high level thought I had reading this.

Reply:
That's fair, though also somewhat intentional - I didn't want this page to become quickly out of sync with what's actually deployed and I also wanted to focus on _what_ we provide (especially in contrast to what vanilla KStreams provides) as opposed to how we accomplish that (dropping hints to the latter) as I think that's what's more interesting from a documentation perspective.